xml/smtp_banners.xml

```diff
@@ -1690,20 +1871,21 @@ The system or service fingerprint with the highest certainty overwrites the othe
     <param pos="3" name="service.version.version"/>
     <param pos="4" name="system.time"/>
   </fingerprint>
-  <fingerprint pattern="^(\S+) E?SMTP Perl" flags="REG_ICASE">
+  <fingerprint pattern="^(?i)(\S+) E?SMTP Perl">
     <description>Some simple PERL SMTP server</description>
     <example host.name="example.com">example.com ESMTP Perl</example>
     <param pos="0" name="service.product" value="Perl"/>
     <param pos="1" name="host.name"/>
   </fingerprint>
-  <fingerprint pattern="^([^ ]+) E?SMTP(?: (?:Service )?Ready\.?)?$" flags="REG_ICASE">
-    <description>
-        catch all for daemons that have no distinguishing fingerprint whatsoever
-      </description>
+  <fingerprint pattern="^(?i)(?:([^ ]+) )?E?SMTP(?: (?:Service )?Ready\.?)?$">
+    <description>catch all for daemons that have no distinguishing fingerprint whatsoever</description>
```

Comment:
I remember there were some concerns for catch-alls.  Specifically around ordering and whether we should include them or not.  @jhart-r7 might have more info here...

Reply:
This catch all isn't new, I've just made the hostname optional for it so that it will matched even shorter non-specific names such as 'ESMTP ready'

Reply:
This may be labeled as a catch all but the regular expression is not.  It looks like it is intended to only match things that have an optional hostname followed by the string "ESMTP Service Ready" and a few minor variants.  A catch-all would do something like match "ESMTP Service Ready" with arbitrary text before and after.